### ISSUE TITLE
No name shown in 'replying to' section of post when using Mitda

### ISSUE THREAD

**Issue by watamelonman** [captainepoch/husky#75]
## General description of the issue

When using Husky with an account on a Mitra instance, replies do not show the target of said reply. There is only blank space next to the 'replying to' section. This blank space happens on all posts regardless of origin (e.g. whether it's from a Pleroma instance, Mastodon, Misskey, Mitra, etc.). This isn't a problem viewing from other fediverse web frontends.
![Screenshot_20240521-114353](https://github.com/captainepoch/husky/assets/170447119/008fa159-0f7c-40d4-b8a7-daa413feb260)

## Husky information

- Husky version: 1.5.4
- Android version: 14
- Android ROM version: CalyxOS 5.7.0

## Mitra version of instance

- Mitra v2.18

## Steps to reproduce it

Please, describe the steps to reproduce the issue:

1. Log into Mitra instance account on Husky
2. View any post that is a reply

## Where did you get Husky from:

- [X] F-Droid.
- [ ] Self-releases.
- [ ] Play Store.
- [ ] Self-compiled (specify tag/commit).

**Comment by captainepoch** [comment]
If Mitra implements the ActivityPub API as Pleroma or Mastodon does, it should show it. I'll take a look, but it's probably not Husky's fault.

**Comment by silverpill** [comment]
@captainepoch Sure, looks like a problem is on the Mitra side. Could you tell where data for this UI element comes from?
I'd expect it to be `Status.in_reply_to_id` or `Status.in_reply_to_account_id`, but Mitra provides both, so it must be something else?

**Comment by captainepoch** [comment]
If it provides any of those (but I think the correct one is `in_reply_to_account_id`), it should show.

Nevertheless, I still have my account in your Mitra instance, so I'll take a look.

**Comment by captainepoch** [comment]
As far as I can see, `in_reply_to_account_id` is null (see attached image).

<img width="570" alt="Captura de Pantalla 2024-05-28 a las 20 20 49" src="https://github.com/captainepoch/husky/assets/1268590/5ce1a347-c25f-44e3-869b-521f08a80bfb">

**Comment by silverpill** [comment]
Yes, but on this image `in_reply_to_id` is also null, so it is probably not a reply?

**Comment by captainepoch** [comment]
You're right about that, I made the screenshot about the wrong value. It is, indeed, with information about the account.

It seems, in the end, a Husky issue, so I'll do more research on why this is happening.

<img width="1131" alt="Captura de Pantalla 2024-05-28 a las 22 27 16" src="https://github.com/captainepoch/husky/assets/1268590/a2a4684e-3889-4911-9512-0bbc1f55796c">

**Comment by captainepoch** [comment]
I didn't forget about this, I'm still investigating why it's happening.

**Comment by captainepoch** [comment]
Hello!

I finally detected what's the issue is here.

Pleroma and Mastodon (I globalize also their forks) include the username in the reply's response Json by the field `in_reply_to_account_acct`. If the user is at the same instance, it doesn't show the `@stereophonic.space` (as an example), and if it's not at the same instance, it does (see screenshots attached).

Mitra should do the same.
<img width="550" alt="Captura de Pantalla 2024-07-10 a las 19 38 49" src="https://github.com/captainepoch/husky/assets/1268590/6a1c9ba2-0910-48aa-9a37-cd159824be13">

<img width="555" alt="Captura de Pantalla 2024-07-10 a las 19 39 20" src="https://github.com/captainepoch/husky/assets/1268590/e2643b86-a56e-44ae-9374-5ffbfe6ec7ce">

**Comment by silverpill** [comment]
I haven't found this field in Mastodon responses, but Pleroma includes `pleroma.in_reply_to_account_acct` and I added it too.
@watamelonman said that fixed the problem, so we can close the issue now.

Thank you!

**Comment by captainepoch** [comment]
I tested this with Mitra 4.9.0 and Husky 1.7.0-beta1, I think it works now:

<img width="1080" height="2400" alt="Image" src="https://github.com/user-attachments/assets/d7a2fc01-d24a-4355-896c-d102f607ff1b" />

Could you let me know?

**Comment by captainepoch** [comment]
Closing it, then. Cheers!
